lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -208,7 +220,21 @@ def add_included_resources_for(serializer, include_tree, primary_data, included)
           end
         end
 
-        def links_for(serializer, options)
+        def links_for(serializer)
+          serializer.links.each_with_object({}) do |(name, value), hash|
+            hash[name] =
+              if value.respond_to?(:call)
```

Comment:
why check respond_to? call?

I think `.is_a?(Proc)` reads better, and makes more sense about the intention of what's going on. Even if there are other objects that you want to support that implement `call`, it would be better to explicitly say what those objects are.

cause, what if I set value to an instance of `Phone` a phone can `call` but it's not the same kind of call.